quickstarts/microsoft.digitaltwins/digitaltwins-with-function-time-series-database-connection/modules/roleassignment.bicep

```diff
@@ -0,0 +1,67 @@
+@description('Existing Digital Twin resource name')
+param digitalTwinsName string
+@description('Existing Event Hubs namespace resource name')
+param eventHubsNamespaceName string
+@description('Existing event hub name')
+param eventHubName string
+@description('Existing Azure Data Explorer cluster resource name')
+param adxClusterName string
+@description('Existing database name')
+param databaseName string
+@description('The id that will be given data owner permission for the Digital Twins resource')
+param principalId string
+@description('The type of the given principal id')
+param principalType string
```

Comment:
isn't this always going to be a servicePrincipal - I didn't see anywhere the user could change it...

Reply:
No, that assumption is not correct. The type could be "User". The purpose of the field is to provide the specific type of the principal ID just data owner assignment to the digital twin resource. In most cases the type would be User, so that they can access and interact with the UI of the digital twin resource.